src/manifest.rs

```diff
@@ -94,6 +94,23 @@ impl Manifest {
 
         sections
     }
+
+    /// returns features exposed by this manifest
+    pub fn features(&self) -> Vec<String> {
+        match self.data.as_table().get("features") {
+            None => vec![],
+            Some(item) => match item {
+                toml_edit::Item::None => vec![],
+                toml_edit::Item::Table(t) => t
+                    .get_values()
+                    .iter()
+                    .map(|(keys, _val)| keys.iter().map(|k| k.to_string()))
+                    .flatten()
+                    .collect(),
+                _ => unreachable!(),
```

Comment:
While other steps in the process should do error handling for us, we should have a better error response (whether to ignore or report error) than `unreachable` without a description.